src/components/InputCheckbox.tsx

```diff
@@ -1,51 +1,70 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Message from './Message';
-interface InputCheckboxProps extends React.InputHTMLAttributes<HTMLInputElement> {
+
+interface InputCheckboxProps
+  extends React.InputHTMLAttributes<HTMLInputElement> {
   label: string | React.ReactNode;
   checked: boolean;
   setChecked: (checked: boolean) => void;
   isValid?: boolean;
-  setIsValid?: (isValid: boolean) => void;
-  validation?: (value: boolean) => boolean;
-  validationMessage?: string;
+  errorMessage?: string;
   showMessage?: boolean;
   isRequired?: boolean;
   res?: any;
-};
+}
 
 const InputCheckbox = ({
-  id, name, label, checked, setChecked,
-  isValid, setIsValid,
-  validation, validationMessage, showMessage,
+  id,
+  name,
+  label,
+  checked,
+  setChecked,
+  isValid,
+  errorMessage,
+  showMessage,
   isRequired,
   ...res
 }: InputCheckboxProps) => {
-  const handleClick = (event:React.FormEvent<HTMLButtonElement>) => {
+
+  const [isTouched, setIsTouched] = useState(false);
+
+  useEffect(() => {
+    checked && setIsTouched(true);
+  }, [checked]);
+
+  const handleClick = (event: React.FormEvent<HTMLButtonElement>) => {
     setChecked(!checked);
-    if (validation) {
-      setIsValid?.(validation(!checked));
-    }
-  }
+  };
+
+  console.log('InputCheckbox', {
+    name,
+    isValid,
+  });
 
   return (
-    <button className={`checkbox-wrapper ${isValid ? 'valid' : ''}`} onClick={handleClick}>
-        <input
-          className='checkbox'
-          id={id}
-          type='checkbox'
-          name={name}
-          checked={checked}
-          defaultChecked={checked}
-          {...res}
-        />
-        <label className='checkbox-label'>{label}</label>
-      <Message
-        type='error'
-        message={validationMessage}
-        showMessage={showMessage && !isValid}
+    <button
+      className={`checkbox-wrapper ${isValid ? 'valid' : ''}`}
+      onClick={handleClick}
+    >
+      <input
+        className='checkbox'
+        id={id}
+        type='checkbox'
+        name={name}
+        checked={checked}
+        readOnly
+        {...res}
       />
+      <label className='checkbox-label'>{label}</label>
```

Comment:
오 그러게요 체크박스에 readonly 속성을 주어 버튼을 통해서만 체크가 가능하도록 제어하기 위한 것이겠죠? 이런 해결방법을 찾으신 것이 대단합니다!!

Reply:
checked 속성에 리액트 상태값을 넘기니까 브라우저 콘솔에 이런 워닝이 뜨더라구요.. 
```
Warning: You provided a `checked` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultChecked`. Otherwise, set either `onChange` or `readOnly`.
```

하라는 데로 추가했는데 저게 베스트인지 모르겠습니다